chain/types/tipset_selector_test.go

```diff
@@ -79,3 +81,156 @@ func TestTipSetSelector_Marshalling(t *testing.T) {
 		})
 	}
 }
+
+func TestTipSetLimit_Marshalling(t *testing.T) {
+	for _, test := range []struct {
+		name          string
+		jsonInput     string
+		expectedLimit types.TipSetLimit
+		expectError   bool
+	}{
+		{
+			name:          "height only",
+			jsonInput:     `{"height": 123}`,
+			expectedLimit: types.TipSetLimits.Height(123),
+		},
+		{
+			name:          "distance only",
+			jsonInput:     `{"distance": 456}`,
+			expectedLimit: types.TipSetLimits.Distance(456),
+		},
+		{
+			name:        "both height and distance",
+			jsonInput:   `{"height": 123, "distance": 456}`,
+			expectError: true,
+		},
+		{
+			name:          "empty object",
+			jsonInput:     `{}`,
+			expectedLimit: types.TipSetLimits.Unlimited,
```

Comment:
I'm not up on other conventions with the API but would it be more clear to have this be more self documenting?

`{"distance":"unlimited"}`

Reply:
That would make `distance` a mixed type field which I think would be more trouble than it's worth.

Reply:
Resolving as tipset limit is now removed from this PR. Happy to discuss this further if/when it is re-introduced.